Esri/arcgis-maps-sdk-swift-samples
# [Fix] Fix info plist and various warnings

## Description

This PR fixes 3 things

- [[SceneConfiguration] Info.plist contained no UIScene configuration dictionary (looking for configuration named "(no name)")](https://developer.apple.com/forums/thread/721912)
    - To make console output look cleaner
- Warning: Result of 'try?' is unused
    - Apple Swift bug [SR-3082](https://bugs.swift.org/browse/SR-3082)
    - https://github.com/apple/swift/issues/45672
    - Instead of having a lingering warning when a user builds the project, suppress it to make the project look better. We can always revisit it once the bug is fixed by Apple
- Move the Samples.entitlements under macOS group
    - Because this file is only used by Mac Catalyst currently
    - Close [card](https://github.com/Esri/arcgis-maps-sdk-swift-samples/projects/1#card-88703883)
    - Also, it makes people feel like the project supports macOS, which is a great keyword to show off at the project's top level.

## How To Test

- Build the app to both iOS and Mac Catalyst
- During the building process, you won't see "try" related warnings anymore
- When the app is running…
    - on `v.next`, it will print "SceneConfiguration" warning into the console
    - this PR will show a clean console. Tap into `Display map` sample and you'll see the first console output about Metal.
